File: sdk/php/config/ResponseConfig.php
<?php
/******************************************************************************
 *
 *  Copyright 2023 Paphus Solutions Inc.
 *
 *  Licensed under the Eclipse Public License, Version 1.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.eclipse.org/legal/epl-v10.html
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 ******************************************************************************/
class ResponseConfig extends Config {
    public ?String $questionId;
	public ?String $responseId;
	public ?String $question;
	public ?String $response;
	public ?String $previous;
	public ?String $onRepeat;
	public ?String $label;
	public ?String $topic;
	public ?String $keywords;
	public ?String $required;
	public ?String $emotions;
	public ?String $actions;
	public ?String $poses;
	public $noRepeat;
	public $requirePrevious;
	public $requireTopic;
	public $flagged;
	public String $correctness;
	public String $command;

    public function toXML() : String {
		$writer = "";
		$this->writeXML($writer);
		return $writer;
	}

    public function writeXML(&$writer) : void {
        $writer .= "<response";
        $this->writeCredentails($writer);
        if(isset($this->questionId)) {
            $writer .= " questionId=\"" . $this->questionId . "\"";
        }
        if(isset($this->responseId)) {
            $writer .= " responseId=\"" . $this->responseId . "\"";
        }
		if(isset($this->label)) {
            $writer .= " label=\"" . $this->label . "\"";
        }
		if(isset($this->topic)) {
            $writer .= " topic=\"" . $this->topic . "\"";
        }
		if(isset($this->keywords)) {
            $writer .= " keywords=\"" . $this->keywords . "\"";
        }
		if(isset($this->required)) {
            $writer .= " required=\"" . $this->required . "\"";
        }
		if(isset($this->emotions)) {
            $writer .= " emotions=\"" . $this->emotions . "\"";
        }
		if(isset($this->actions)) {
            $writer .= " actions=\"" . $this->actions . "\"";
        }
		if(isset($this->poses)) {
            $writer .= " poses=\"" . $this->poses . "\"";
        }
		if(isset($this->correctness)) {
            $writer .= " correctness=\"" . $this->correctness . "\"";
        }
		$writer .= " noRepeat=\"" . $this->noRepeat . "\"";
		$writer .= " requirePrevious=\"" . $this->requirePrevious . "\"";
		$writer .= " requireTopic=\"" . $this->requireTopic . "\"";
		$writer .= " flagged=\"" . $this->flagged . "\"";
		$writer .= ">";
		if(isset($this->question)) {
			$writer .= "<question>";
			$writer .= Utils::escapeHTML($this->question);
			$writer .= "</question>";
		}
		if(isset($this->response)) {
			$writer .= "<response>";
			$writer .= Utils::escapeHTML($this->response);
			$writer .= "</response>";
		}
		if(isset($this->previous)) {
			$writer .= "<previous>";
			$writer .= Utils::escapeHTML($this->previous);
			$writer .= "</previous>";
		}
		if(isset($this->onRepeat)) {
			$writer .= "<onRepeat>";
			$writer .= Utils::escapeHTML($this->onRepeat);
			$writer .= "</onRepeat>";
		}
		if(isset($this->command)) {
			$writer .= "<command>";
			$writer .= Utils::escapeHTML($this->command);
			$writer .= "</command>";
		}
		$writer .= "</response>";
    }

    public function parseXML($xml) : void {
		$xmlData = Utils::loadXML($xml);
		if($xmlData===false) {
			return;
		}
        $this->questionId = $xmlData->attributes()->questionId;
        $this->responseId = $xmlData->attributes()->responseId;
        $this->label = $xmlData->attributes()->label;
		$this->topic = $xmlData->attributes()->topic;
		$this->keywords = $xmlData->attributes()->keywords;
		$this->required = $xmlData->attributes()->required;
		$this->emotions = $xmlData->attributes()->emotions;
        $this->actions = $xmlData->attributes()->actions;
        $this->poses = $xmlData->attributes()->poses;
		$this->type = $xmlData->attributes()->type;
		$this->correctness = $xmlData->attributes()->correctness;
		if(isset($xmlData->attributes()->noRepeat)) {
			$this->noRepeat =(bool) $xmlData->attributes()->noRepeat;
		}
		if(isset($xmlData->attributes()->requireTopic)) {
			$this->requireTopic = (bool) $xmlData->attributes()->requireTopic;
		}

		if(isset($xmlData->attributes()->flagged)) {
			$this->flagged = (bool) $xmlData->attributes()->flagged;
		}

		if(isset($xmlData->attributes()->requirePrevious)) {
			$this->requirePrevious =(bool)  $xmlData->attributes()->requirePrevious;
		}
		if(isset($xmlData->question)) {
			$this->question = $xmlData->question;
		}
		if(isset($xmlData->response)) {
			$this->response = $xmlData->response;
		}
		if(isset($xmlData->question)) {
			$this->question = $xmlData->question;
		}
		if(isset($xmlData->command)) {
			$this->command = $xmlData->command;
			$this->command = str_replace('&#34;', '"', $this->command);
		}

    }
}
?>

File: sdk/python/config/ResponseConfig.py
############################################################################
#
#  Copyright 2023 Paphus Solutions Inc.
#
#  Licensed under the Eclipse Public License, Version 1.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#      http://www.eclipse.org/legal/epl-v10.html
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.
#
############################################################################
from config.Config import Config
from util.Utils import Writer, Utils


class ResponseConfig(Config):
    questionId: str
    responseId: str
    question: str
    response: str
    previous: str
    onRepeat: str
    label: str
    topic: str
    keywords: str
    required: str
    emotions: str
    actions: str
    poses: str
    noRepeat: bool
    requirePrevious: bool
    requireTopic: bool
    flagged: bool
    correctness: str
    command: str

    def __init__(self):
        self.questionId = None
        self.responseId = None
        self.question = None
        self.response = None
        self.previous = None
        self.onRepeat = None
        self.label = None
        self.topic = None
        self.keywords = None
        self.required = None
        self.emotions = None
        self.actions = None
        self.poses = None
        self.noRepeat = None
        self.requirePrevious = None
        self.requireTopic = None
        self.flagged = None
        self.correctness = None
        self.command = None

    def toXML(self):
        writer = Writer()
        self.writerXML(writer)
        return writer

    def writerXML(self, writer: Writer):
        writer = Writer("<response")
        self.writeCredentials(writer)
        if(self.questionId != None):
            writer.append(" questionId=\"" + self.questionId + "\"")
        if(self.responseId != None):
            writer.append(" responseId=\"" + self.responseId + "\"")
        if(self.label != None):
            writer.append(" label=\"" + self.label + "\"")
        if(self.topic != None):
            writer.append(" topic=\"" + self.topic + "\"")
        if(self.keywords != None):
            writer.append(" keywords=\"" + self.keywords + "\"")
        if(self.required != None):
            writer.append(" required=\"" + self.required + "\"")
        if(self.emotions != None):
            writer.append(" emotions=\"" + self.emotions + "\"")
        if(self.actions != None):
            writer.append(" actions=\"" + self.actions + "\"")
        if(self.poses != None):
            writer.append(" poses=\"" + self.poses + "\"")
        if(self.correctness != None):
            writer.append(" correctness=\"" + self.correctness + "\"")

        writer.append(" noRepeat=\"true\"") if self.noRepeat == True else None
        writer.append(" requirePrevious=\"true\"") if self.requirePrevious == True else None
        writer.append(" requireTopic=\"true\"") if self.requireTopic == True else None
        writer.append(" flagged=\"true\"") if self.flagged == True else None
        writer.append(">")
        if(self.question != None):
            writer.append("<question>")
            writer.append(Utils.escapeHTML(self.question))
            writer.append("</question>")

        if(self.response != None):
            writer.append("<response>")
            writer.append(Utils.escapeHTML(self.response))
            writer.append("</response>")

        if(self.previous != None):
            writer.append("<previous>")
            writer.append(Utils.escapeHTML(self.previous))
            writer.append("</previous>")

        if(self.onRepeat != None):
            writer.append("<onRepeat>")
            writer.append(Utils.escapeHTML(self.onRepeat))
            writer.append("</onRepeat>")

        if(self.command != None):
            writer.append("<command>")
            writer.append(Utils.escapeHTML(self.command))
            writer.append("</command>")
        writer.append("</response>")

    def parseXML(self, xml:str):
        super().parseXML(xml)
        root = Utils.loadXML(xml)
        if(root == None):
            return
        self.questionId = root.attrib.get("questionId")
        self.responseId = root.attrib.get("responseId")
        self.label = root.attrib.get("label")
        self.topic = root.attrib.get("topic")
        self.keywords = root.attrib.get("keywords")
        self.required = root.attrib.get("required")
        self.emotions = root.attrib.get("emotions")
        self.actions = root.attrib.get("actions")
        self.poses = root.attrib.get("poses")
        self.type = root.attrib.get("type")
        self.correctness = root.attrib.get("correctness")
        # tag name
        if(root.find("noRepeat") != None):
            self.noRepeat = root.find("noRepeat")
        if(root.find("requireTopic") != None):
            self.requireTopic = root.find("requireTopic")
        if(root.find("flagged") != None):
            self.flagged = root.find("flagged")
        if(root.find("requirePrevious") != None):
            self.requirePrevious = root.find("requirePrevious")
        if(root.find("question") != None):
            self.question = root.find("question")
        if(root.find("response") !=None):
            self.response = root.find("response")

        if(root.find("command") != None):
            self.command = root.find("command")
            self.command = str(self.command).replace("&#34;", "\"")
